src/distances.jl

```diff
@@ -24,64 +84,47 @@ julia> Ellipsoidal([1.0,0.5], [π/2])
 julia> Ellipsoidal([1.0,0.5,0.5], [π/2,0.0,0.0])
 ```
 """
+
 struct Ellipsoidal{N,T} <: Metric
   dist::Mahalanobis{T}
 
-  function Ellipsoidal{N,T}(semiaxes, angles) where {N,T}
+  function Ellipsoidal{N,T}(semiaxes, angles; convention=:TaitBryanExtr) where {N,T}
     @assert length(semiaxes) == N "number of semiaxes must match spatial dimension"
     @assert all(semiaxes .> zero(T)) "semiaxes must be positive"
     @assert N ∈ [2,3] "dimension must be either 2 or 3"
 
-    # scaling matrix
-    Λ = Diagonal(one(T)./semiaxes.^2)
-
-    # rotation matrix
-    if N == 2
-      θ = angles[1]
-
-      cosθ = cos(θ)
-      sinθ = sin(θ)
+    rule = rules[convention]
 
-      P = [cosθ -sinθ
-           sinθ  cosθ]
+    # invert x and y if necessary
+    if rule.main == :y
+       semiaxes[1], semiaxes[2] = semiaxes[2], semiaxes[1]
     end
-    if N == 3
-      θxy, θyz, θzx = angles
 
-      cosxy = cos(θxy)
-      sinxy = sin(θxy)
-      cosyz = cos(θyz)
-      sinyz = sin(θyz)
-      coszx = cos(θzx)
-      sinzx = sin(θzx)
-
-      _1 = one(T)
-      _0 = zero(T)
-
-      Rxy = [cosxy -sinxy _0
-             sinxy  cosxy _0
-             _0     _0 _1]
-
-      Ryz = [_1    _0     _0
-             _0 cosyz -sinyz
-             _0 sinyz  cosyz]
+    # scaling matrix
+    Λ = Diagonal(one(T)./semiaxes.^2)
+    #Λ = SMatrix{N,N}(Diagonal(one(T)./semiaxes.^2))
 
-      Rzx = [ coszx _0 sinzx
-             _0 _1    _0
-             -sinzx _0 coszx]
+    # convert to radian and invert sign if necessary
+    !rule.radian && (angles = deg2rad.(angles))
+    _0 = zero(eltype(angles))
+    N == 2 && (angles = [angles[1], _0, _0])
+    intr = @. (rule.motion == :CW) & !rule.extrinsic
+    extr = @. (rule.motion == :CCW) & rule.extrinsic
+    angles[intr .| extr] *= -1
 
-      P = Rzx*Ryz*Rxy
-    end
+    # rotation matrix
+    P = angle_to_dcm(angles..., rule.order)[1:N,1:N]
+    #P = SMatrix{N,N}(angle_to_dcm(angles..., rule.order)[1:N,1:N])
```

Comment:
I think angle_to_dcm already returns a static matrix?

Reply:
Yes. But when I do [1:N,1:N] it seems to convert back to `Matrix`

Reply:
Wow, that is surprising. I wonder if we should ask it on Zulip. This behavior is certainly strange.

Reply:
Currently we have to do `[SOneTo(N),SOneTo(N)]` to get a static array back. Could you adapt it here?

Source: https://discourse.julialang.org/t/recent-developments-in-staticarrays-jl-land/47778/6?u=juliohm

Reply:
Done. All working locally. Just need to wait the new version of Distances.jl to work properly for all